Dockerfile

```
@@ -1,9 +1,6 @@
 FROM alpine:3.17
 
-RUN apk update && \
-    apk upgrade && \
-    apk add bash procps drill git coreutils libidn curl socat openssl xxd && \
-    rm -rf /var/cache/apk/* && \
+RUN apk --no-cache --upgrade add bash procps drill git coreutils libidn curl socat openssl xxd && \
```

Alternative if you want to upgrade all packages prior to adding new ones:

```suggestion
RUN apk --no-cache upgrade && \
    apk --no-cache add bash procps drill git coreutils libidn curl socat openssl xxd && \
```

---

A recent PR by @jauderho that was closed also tried to include the `-a` (`--available`) option to the `apk upgrade` command.
- This forces upgrades on installed packages that don't have new versions, which [Alpine notes sometimes is required](https://wiki.alpinelinux.org/wiki/Upgrading_Alpine) when upgrading.
- However, [Alpine switched from `uClibc` to `musl` back in 2014](https://en.wikipedia.org/wiki/Alpine_Linux), so those official docs seem a bit outdated.

As such, it's not something I'd bother with, but if you do still want this separate upgrade step, here's the same suggestion as above, but with `--available`:

Alternative if you want to upgrade all packages prior to adding new ones:

```suggestion
RUN apk --no-cache upgrade --available && \
    apk --no-cache add bash procps drill git coreutils libidn curl socat openssl xxd && \
```